sync/class.jetpack-sync-module-users.php

```diff
@@ -292,4 +322,70 @@ public function expand_users( $args ) {
 
 		return array_map( array( $this, 'sanitize_user_and_expand' ), get_users( array( 'include' => $user_ids ) ) );
 	}
+
+	public function remove_user_from_blog_handler( $user_id, $blog_id ) {
+		//User is removed on add, see https://github.com/WordPress/WordPress/blob/0401cee8b36df3def8e807dd766adc02b359dfaf/wp-includes/ms-functions.php#L2114
+		if ( $this->is_add_new_user_to_blog() ) {
+			return;
+		}
+
+		$reassigned_user_id = $this->get_reassigned_network_user_id();
+
+		//Note that we are in the context of the blog the user is removed from, see https://github.com/WordPress/WordPress/blob/473e1ba73bc5c18c72d7f288447503713d518790/wp-includes/ms-functions.php#L233
+		do_action( 'jetpack_removed_user_from_blog', $user_id, $reassigned_user_id );
+	}
+
+	private function is_add_new_user_to_blog() {
+		return $this->is_function_in_backtrace( 'add_new_user_to_blog' );
+	}
+
+	private function is_create_user() {
+		$functions = array(
+			'add_new_user_to_blog', // Used to suppress jetpack_sync_save_user in save_user_cap_handler when user registered
+			'wp_create_user' // Used to suppress jetpack_sync_save_user in save_user_role_handler when user registered
+		);
+
+		return $this->is_function_in_backtrace( $functions );
+	}
+
+	private function get_reassigned_network_user_id() {
+		$backtrace = debug_backtrace( false );
+		foreach ( $backtrace as $call ) {
+			if (
+				'remove_user_from_blog' === $call['function'] &&
+				3 === count( $call['args'] )
+			) {
+				return $call['args'][2];
+			}
+		}
+
+		return false;
+	}
+
+	private function is_function_in_backtrace( $names ) {
```

Comment:
this method seems pretty handy, maybe we could move it to `Jetpack:: is_function_in_backtrace`?